### PR TITLE
add recommendation Kyverno setup

### LINKED ISSUE #559: [Enhancement] Kyverno recommendation setup
### Description

Kyverno recommended configurations on different scales of clusters

### Slack discussion

_No response_

### PR DESCRIPTION
**Pull request by husnialhamdani**
## Related issue #
Closes #559

## Proposed Changes
Kyverno recommended configurations on different scales of clusters

## Checklist

- [X] I have read the [contributing guidelines](https://github.com/kyverno/website/blob/main/CONTRIBUTING.md).
- [X] I have inspected the website preview for accuracy.
- [X] I have signed off my issue.

### CODE REVIEW

**Comment by husnialhamdani** [comment]
> @husnialhamdani you must sign off on all of your commits for DCO



> @husnialhamdani you must sign off on all of your commits for DCO

signed

**Review by samj1912 — APPROVED** [review]

File: content/en/docs/Installation/_index.md [hunk 1]
```diff
@@ -644,3 +644,18 @@ kubectl delete mutatingwebhookconfigurations kyverno-policy-mutating-webhook-cfg
...
+* `less than 4000 total resource`: use default limit
+* `more than 4000 total resource`: use 768Mi (2x the default)
+### Kyverno mode:
+Recommendation setup based on environment
+#### Environment:
+* `Staging/demo`: standalone mode (1 replicas)
+* `Production`: High Availability mode (3 replicas)
+
+{{% alert title="Note" color="info" %}}
+total resources : combination of static (ConfigMap, Secret, etc.) and workloads (Pod, Cronjob, Deployment, etc.)
```
Comment (JimBugwadia) [inline content/en/docs/Installation/_index.md:660]: Are all these resources cached by Kyverno and hence will impact memory?

Perhaps we should add a separate section called "Resource Caching" and define the list there. Then in the memory configuration reference as "cached resources".

File: content/en/docs/Installation/_index.md [hunk 2]
```diff
@@ -644,3 +644,18 @@ kubectl delete mutatingwebhookconfigurations kyverno-policy-mutating-webhook-cfg
 
 kubectl delete validatingwebhookconfigurations kyverno-policy-validating-webhook-cfg kyverno-resource-validating-webhook-cfg
 ```
+## Recommended setup
+Kyverno recommended configurations on different scales of clusters
+
+### Kyverno Memory limit config:
```
Comment (JimBugwadia) [inline content/en/docs/Installation/_index.md:650]: Use title case

File: content/en/docs/Installation/_index.md [hunk 3]
```diff
@@ -644,3 +644,18 @@ kubectl delete mutatingwebhookconfigurations kyverno-policy-mutating-webhook-cfg
...
+* `less than 4000 total resource`: use default limit
+* `more than 4000 total resource`: use 768Mi (2x the default)
```
Comment (JimBugwadia) [inline content/en/docs/Installation/_index.md:652]: What about greater than 4K?

**Review by JimBugwadia — CHANGES_REQUESTED** [review]
This will be a great help! Minor comments.

File: content/en/docs/Installation/_index.md [hunk 4]
```diff
@@ -644,3 +644,18 @@ kubectl delete mutatingwebhookconfigurations kyverno-policy-mutating-webhook-cfg
...
+### Kyverno mode:
```
Comment (JimBugwadia) [inline content/en/docs/Installation/_index.md:653]: Do we need this here? Don't we have HA covered elsewhere?

File: content/en/docs/Installation/_index.md [hunk 5]
```diff
@@ -644,3 +644,18 @@ kubectl delete mutatingwebhookconfigurations kyverno-policy-mutating-webhook-cfg
 
 kubectl delete validatingwebhookconfigurations kyverno-policy-validating-webhook-cfg kyverno-resource-validating-webhook-cfg
 ```
+## Recommended setup
+Kyverno recommended configurations on different scales of clusters
```
Comment (JimBugwadia) [inline content/en/docs/Installation/_index.md:648]: Should we explain how we are testing or measuring the data?

File: content/en/docs/Installation/_index.md (hunk 4 above)
Comment (chipzoller) [reply content/en/docs/Installation/_index.md:653]: We do.

File: content/en/docs/Installation/_index.md (hunk 5 above)
Comment (chipzoller) [inline content/en/docs/Installation/_index.md:648]: New line after all headings.

File: content/en/docs/Installation/_index.md (hunk 3 above)
Comment (chipzoller) [reply content/en/docs/Installation/_index.md:652]: These aren't code and don't need code formatting.

File: content/en/docs/Installation/_index.md (hunk 1 above)
Comment (chipzoller) [reply content/en/docs/Installation/_index.md:660]: Probably best to put this before getting to the config since this is how those numbers are defined.

**Review by chipzoller — CHANGES_REQUESTED** [review]

File: content/en/docs/Installation/_index.md [hunk 6]
```diff
@@ -644,3 +644,19 @@ kubectl delete mutatingwebhookconfigurations kyverno-policy-mutating-webhook-cfg
 
 kubectl delete validatingwebhookconfigurations kyverno-policy-validating-webhook-cfg kyverno-resource-validating-webhook-cfg
 ```
+## Recommended setup
+
+Kyverno recommended configurations on different scales of clusters
+
+### Kyverno Memory limit config:
+* less than 4000 total resource: use default limit
```
Comment (chipzoller) [inline content/en/docs/Installation/_index.md:652]: New line.

File: content/en/docs/Installation/_index.md [hunk 7]
```diff
@@ -644,3 +644,19 @@ kubectl delete mutatingwebhookconfigurations kyverno-policy-mutating-webhook-cfg
...
+* more than 4000 total resource: use 768Mi (2x the default)
+### Kyverno mode:
+Recommendation setup based on environment
+#### Environment:
```
Comment (chipzoller) [inline content/en/docs/Installation/_index.md:656]: New line.

File: content/en/docs/Installation/_index.md [hunk 8]
```diff
@@ -644,3 +644,19 @@ kubectl delete mutatingwebhookconfigurations kyverno-policy-mutating-webhook-cfg
...
+* `Staging/demo`: standalone mode (1 replicas)
```
Comment (chipzoller) [inline content/en/docs/Installation/_index.md:657]: New line.

File: content/en/docs/Installation/_index.md [hunk 9]
```diff
@@ -644,3 +644,19 @@ kubectl delete mutatingwebhookconfigurations kyverno-policy-mutating-webhook-cfg
...
 ```
+## Recommended setup
+
+Kyverno recommended configurations on different scales of clusters
+
+### Kyverno Memory limit config:
+* less than 4000 total resource: use default limit
+* more than 4000 total resource: use 768Mi (2x the default)
+### Kyverno mode:
+Recommendation setup based on environment
```
Comment (chipzoller) [inline content/en/docs/Installation/_index.md:655]: New line.

File: content/en/docs/Installation/_index.md (hunk 3 above)
Comment (husnialhamdani) [reply content/en/docs/Installation/_index.md]: @JimBugwadia actually we want to run the test on greater than 4K resources but our current test server not able to  accommodate that

File: content/en/docs/Installation/_index.md (hunk 4 above)
Comment (husnialhamdani) [reply content/en/docs/Installation/_index.md]: removed the Kyverno mode

File: content/en/docs/Installation/_index.md [hunk 10]
```diff
@@ -644,3 +644,16 @@ kubectl delete mutatingwebhookconfigurations kyverno-policy-mutating-webhook-cfg
...
 kubectl delete validatingwebhookconfigurations kyverno-policy-validating-webhook-cfg kyverno-resource-validating-webhook-cfg
 ```
+## Recommended setup
+
+Kyverno recommended configurations on different scales of clusters
+
+### Kyverno Memory limit config:
+
+* less than 4000 total resource: use default limit
+* more than 4000 total resource: use 768Mi (2x the default)
```
Comment (realshuting) [inline content/en/docs/Installation/_index.md:654]: @husnialhamdani - were these data accurate? What was the memory limit during the test?

We may need to update these after the memory leak issue is fixed.

**Comment by chipzoller** [comment]
This PR is still in draft. Is it being actively revised or should it be closed?

**Comment by husnialhamdani** [comment]
> This PR is still in draft. Is it being actively revised or should it be closed?

@realshuting just informed that the leak issue on 1.7 version has been fixed, now I'm re-executing the performance test

**Comment by chipzoller** [comment]
Because this has been sitting around for a bit it's going to get outdated with the changes for 1.8, so let's just close this and have you open another once you're ready.

**Pull request closed without merging**